lib/SILOptimizer/Mandatory/DiagnoseUnreachable.cpp

```diff
@@ -785,6 +890,18 @@ void swift::performSILDiagnoseUnreachable(SILModule *M, SILModuleTransform *T) {
     // Remove unreachable blocks.
     removeUnreachableBlocks(Fn, *M, &State);
 
+    // Gather exit blocks to check for naive self-recursion.
+    llvm::SmallPtrSet<SILBasicBlock*, 4> ExitBlocks;
```

Comment:
IIRC, there was a utility function for something like this somewhere. But may be I'm wrong.

Reply:
There's `SILFunction::findExitingBlocks`, but it doesn't look like its other usage in `Local.cpp` depends on it being ordered.  I could switch them both to using a `SmallPtrSet`.

Reply:
Yeah, I think it should be fine to switch them to use `SmallPtrSet `